schema/openapi_parser.go

```diff
@@ -63,20 +64,147 @@ func getKinds[T KindVersion](s *OpenApiParser, strict bool, buildKindVersion fun
 	return result, nil
 }
 
-func (s *OpenApiParser) GetConsoleKinds(strict bool) (map[string]Kind, error) {
+func (s *OpenApiParser) getRuns(backendType BackendType) (RunCatalog, error) {
+	result := make(RunCatalog, 0)
+	for path := s.doc.Model.Paths.PathItems.First(); path != nil; path = path.Next() {
+		err := handleExecuteOperation(backendType, path.Key(), path.Value().Get, resty.MethodGet, result)
+		if err != nil {
+			return nil, err
+		}
+		err = handleExecuteOperation(backendType, path.Key(), path.Value().Post, resty.MethodPost, result)
+		if err != nil {
+			return nil, err
+		}
+		err = handleExecuteOperation(backendType, path.Key(), path.Value().Put, resty.MethodPut, result)
+		if err != nil {
+			return nil, err
+		}
+		err = handleExecuteOperation(backendType, path.Key(), path.Value().Delete, resty.MethodDelete, result)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return result, nil
+}
+
+func handleExecuteOperation(backendType BackendType, path string, operation *v3high.Operation, method string, result RunCatalog) error {
+	if operation == nil {
+		return nil
+	}
+
+	nameYaml, present := operation.Extensions.Get("x-cdk-run-name")
+	if !present {
+		return nil
+	}
+	name := nameYaml.Value
+	docYaml, docPresent := operation.Extensions.Get("x-cdk-run-doc")
+	var doc string
+	if docPresent {
+		doc = docYaml.Value
+	} else {
+		doc = ""
+	}
+	run := Run{
+		BackendType:    backendType,
+		Path:           path,
+		Name:           name,
+		Doc:            doc,
+		QueryParameter: make(map[string]FlagParameterOption, len(operation.Parameters)),
+		PathParameter:  make([]string, 0, len(operation.Parameters)),
```

Comment:
I think the init size might be wrong, here we want to count the number of parameters of type query and path separatly no ?
As it is now, it will init a map of the addtion of path and query parametes count as I undersand it.

Reply:
```
A slice can be created with the built-in function called make, which has the signature,

func make([]T, len, cap) []T
```
[src](https://go.dev/blog/slices-intro)

Therefore here the init size is 0, it's just the underlying backend array that is init to `len(operation.Parameters)`